### PR TITLE
feat(corpusItem): adding in savedItem on corpus item

### PR DESCRIPTION
**Pull request by bassrock**
## Goal

Clients need to know if a Corpus Item has been saved to a user's saves for display purposes. (ie. should they display a saved button on recomendations)

### CODE REVIEW

File: schema.graphql [hunk 1]
```diff
@@ -1,3 +1,5 @@
+extend schema @link(url: "https://specs.apollo.dev/federation/v2.0", import: ["@key", "@external"])
```
Comment (bassrock) [inline schema.graphql:1]: switching to fed 2.0 for cleaner syntax

File: schema.graphql [hunk 2]
```diff
@@ -460,7 +471,7 @@ type SavedItemTagAssociation {
   tagId: ID!
 }
 
-extend type User @key(fields: "id") {
+type User @key(fields: "id") {
```
Comment (bassrock) [inline schema.graphql:474]: fed 2 drops extend syntax

File: src/resolvers/index.ts [hunk 3]
```diff
@@ -39,13 +39,7 @@ const resolvers = {
   },
   Item: {
     savedItem: async (item: Item, args, context: IContext) => {
-      const save = await context.dataLoaders.savedItemsByUrl.load(
-        item.givenUrl
-      );
-      if (save == null) {
-        throw new NotFoundError(`No Save found for url=${item.givenUrl}`);
```
Comment (bassrock) [inline src/resolvers/index.ts:46]: we shouldnt be throwing a not found error here. we should just return null, and continue on. There are plent of cases where an Item won't return a savedItem and its not an error

**Review by kschelonka — DISMISSED** [review]

File: src/test/graphql/queries/item.integration.ts [hunk 4]
```diff
@@ -173,7 +173,7 @@ describe('item', () => {
     expect(entities.length).toEqual(1);
     expect(entities[0]).toEqual(expected);
   });
-  it('returns null with not found error if the save does not exist', async () => {
+  it('returns null if the save does not exist', async () => {
```
Comment (kschelonka) [inline src/test/graphql/queries/item.integration.ts:176]: serious deja vu here, I thought we already did this :| but yes I agree

**Pull request merged**